app/src/main/java/com/example/rickandmortywithcash/koin/ServiceModule.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.rickandmortywithcash.koin
+
+import androidx.room.Room
+import com.example.rickandmortywithcash.Constants
+import com.example.rickandmortywithcash.retrofit.RickAndMortyApi
+import com.example.rickandmortywithcash.room.AppDataBase
+import com.example.rickandmortywithcash.screens.viewmodel.ViewModelList
+import com.example.rickandmortywithcash.service.CharacterRepository
+import com.example.rickandmortywithcash.service.Repository
+import com.example.rickandmortywithcash.service.Service
+import com.example.rickandmortywithcash.service.ServiceImpl
+import okhttp3.OkHttpClient
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+val serviceModule = module {
+
+    single {
+        ServiceImpl(
+            api = get(),
+            characterRepository = get()
+        )
+    }
```

Comment:
я так понимаю здесь есть какой-то интерфейс, если используется Impl
```suggestion
    single<Service> {
        ServiceImpl(
            api = get(),
            characterRepository = get()
        )
    }
```
чтобы не работать с конкретной реализацией интерфейса